src/common/board.rs

```diff
@@ -31,39 +45,63 @@ pub(crate) fn register_models(registry: &mut ComponentRegistry) {
     }
 }
 
-pub struct FakeBoard {
-    analogs: Vec<Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>>,
-    i2cs: HashMap<String, Arc<Mutex<FakeI2CHandle>>>,
-    pin_pwms: HashMap<i32, f64>,
-    pin_pwm_freq: HashMap<i32, u64>,
-}
+/// Represents the functionality of a general purpose compute board that contains various components such as analog readers and digital interrupts.
 pub trait Board: Status {
+    /// Set a pin to high or low
     fn set_gpio_pin_level(&mut self, pin: i32, is_high: bool) -> anyhow::Result<()>;
+
+    /// Return the current [BoardStatus] of the board
     fn get_board_status(&self) -> anyhow::Result<common::v1::BoardStatus>;
+
+    /// Get the state of a pin, high(`true`) or low(`false`)
     fn get_gpio_level(&self, pin: i32) -> anyhow::Result<bool>;
+
+    /// Get an [AnalogReader] by name
     fn get_analog_reader_by_name(
         &self,
         name: String,
     ) -> anyhow::Result<Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>>;
+
+    /// Set the board to the indicated [PowerMode]
     fn set_power_mode(
         &self,
-        mode: component::board::v1::PowerMode,
+        mode: PowerMode,
         duration: Option<Duration>,
     ) -> anyhow::Result<()>;
+
```

Comment:
Looks like no doc comment for this function. Per the other discussion about i2c I recognize there may be upcoming changes, but it feels OK to me to document what exists here now, even if it ends up moving elsewhere later.

Reply:
I agree